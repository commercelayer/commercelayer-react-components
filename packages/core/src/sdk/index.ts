import sdk, { type CommerceLayerClient } from "@commercelayer/sdk"
import { jwtDecode } from "jwt-decode"
import type { Jwt, RequestConfig } from "src/types/index.js"

export function getSdk({ accessToken }: RequestConfig): CommerceLayerClient {
  const organization = jwtDecode<Jwt>(accessToken).organization.slug
  return sdk({ accessToken, organization })
}
