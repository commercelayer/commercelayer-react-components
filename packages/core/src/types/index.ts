export type { RequestConfig } from "./base"
