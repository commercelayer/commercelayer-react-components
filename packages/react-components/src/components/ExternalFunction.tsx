import ExternalFunctionContext, {
  callExternalFunction
} from '#context/ExternalFunctionContext'
import { type ReactNode } from 'react'

interface Props {
  children: ReactNode
  url: string
}

export function ExternalFunction({ children, url }: Props): JSX.Element {
  return (
    <ExternalFunctionContext.Provider value={{ url, callExternalFunction }}>
      {children}
    </ExternalFunctionContext.Provider>
  )
}

export default ExternalFunction
