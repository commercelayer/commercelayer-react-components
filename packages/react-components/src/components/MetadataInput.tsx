import BaseInput from '#components/utils/BaseInput'
import { type BaseInputComponentProps } from '#typings'

type Props = BaseInputComponentProps &
  JSX.IntrinsicElements['input'] &
  JSX.IntrinsicElements['textarea']

export function MetadataInput(props: Props): JSX.Element {
  return <BaseInput data-metadata {...props} />
}

export default MetadataInput
