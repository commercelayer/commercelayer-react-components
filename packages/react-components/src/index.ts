export * from '#components/errors/Errors'
export * from '#components/ExternalFunction'
export * from '#components/MetadataInput'
export * from '#components/SubmitButton'
export * from '#components/addresses/Address'
export * from '#components/addresses/AddressCountrySelector'
export * from '#components/addresses/AddressField'
export * from '#components/addresses/AddressInput'
export * from '#components/addresses/AddressStateSelector'
export * from '#components/addresses/AddressesContainer'
export * from '#components/addresses/AddressesEmpty'
export * from '#components/addresses/BillingAddressContainer'
export * from '#components/addresses/BillingAddressForm'
export * from '#components/addresses/SaveAddressesButton'
export * from '#components/addresses/ShippingAddressContainer'
export * from '#components/addresses/ShippingAddressForm'
export * from '#components/auth/CommerceLayer'
export * from '#components/customers/CustomerAddressForm'
export * from '#components/customers/CustomerContainer'
export * from '#components/customers/CustomerField'
export * from '#components/customers/CustomerInput'
export * from '#components/customers/SaveCustomerButton'
export * from '#components/gift_cards/GiftCard'
export * from '#components/gift_cards/GiftCardContainer'
export * from '#components/gift_cards/GiftCardCurrencySelector'
export * from '#components/gift_cards/GiftCardInput'
export * from '#components/gift_cards/GiftCardOrCouponCode'
export * from '#components/gift_cards/GiftCardOrCouponForm'
export * from '#components/gift_cards/GiftCardOrCouponInput'
export * from '#components/gift_cards/GiftCardOrCouponRemoveButton'
export * from '#components/gift_cards/GiftCardOrCouponSubmit'
export * from '#components/line_items/LineItemsContainer'
export * from '#components/line_items/LineItem'
export * from '#components/line_items/LineItemAmount'
export * from '#components/line_items/LineItemCode'
export * from '#components/line_items/LineItemField'
export * from '#components/line_items/LineItemImage'
export * from '#components/line_items/LineItemName'
export * from '#components/line_items/LineItemOption'
export * from '#components/line_items/LineItemOptions'
export * from '#components/line_items/LineItemQuantity'
export * from '#components/line_items/LineItemRemoveLink'
export * from '#components/line_items/LineItemsCount'
export * from '#components/line_items/LineItemsEmpty'
export * from '#components/orders/AddToCartButton'
export * from '#components/orders/AdjustmentAmount'
export * from '#components/orders/CartLink'
export * from '#components/orders/CheckoutLink'
export * from '#components/orders/DiscountAmount'
export * from '#components/orders/GiftCardAmount'
export * from '#components/orders/OrderContainer'
export * from '#components/orders/OrderList'
export * from '#components/orders/OrderListEmpty'
export * from '#components/orders/OrderListRow'
export * from '#components/orders/OrderNumber'
export * from '#components/orders/OrderStorage'
export * from '#components/orders/PaymentMethodAmount'
export * from '#components/orders/PlaceOrderButton'
export * from '#components/orders/PlaceOrderContainer'
export * from '#components/orders/PrivacyAndTermsCheckbox'
export * from '#components/orders/ShippingAmount'
export * from '#components/orders/SubTotalAmount'
export * from '#components/orders/TaxesAmount'
export * from '#components/orders/TotalAmount'
export * from '#components/parcels/ParcelField'
export * from '#components/parcels/ParcelLineItem'
export * from '#components/parcels/ParcelLineItemField'
export * from '#components/parcels/ParcelLineItemsCount'
export * from '#components/parcels/Parcels'
export * from '#components/parcels/ParcelsCount'
export * from '#components/payment_methods/PaymentMethod'
export * from '#components/payment_methods/PaymentMethodName'
export * from '#components/payment_methods/PaymentMethodPrice'
export * from '#components/payment_methods/PaymentMethodRadioButton'
export * from '#components/payment_methods/PaymentMethodsContainer'
export * from '#components/payment_source/PaymentSource'
export * from '#components/payment_source/PaymentSourceBrandIcon'
export * from '#components/payment_source/PaymentSourceBrandName'
export * from '#components/payment_source/PaymentSourceDetail'
export * from '#components/payment_source/PaymentSourceEditButton'
export * from '#components/prices/Price'
export * from '#components/prices/PricesContainer'
export * from '#components/shipments/Shipment'
export * from '#components/shipments/ShipmentField'
export * from '#components/shipments/ShipmentsContainer'
export * from '#components/shipments/ShipmentsCount'
export * from '#components/shipping_methods/ShippingMethod'
export * from '#components/shipping_methods/ShippingMethodName'
export * from '#components/shipping_methods/ShippingMethodPrice'
export * from '#components/shipping_methods/ShippingMethodRadioButton'
export * from '#components/skus/AvailabilityContainer'
export * from '#components/skus/AvailabilityTemplate'
export * from '#components/skus/DeliveryLeadTime'
export * from '#components/skus/SkuField'
export * from '#components/skus/SkuList'
export * from '#components/skus/SkuListsContainer'
export * from '#components/skus/Skus'
export * from '#components/skus/SkusContainer'
export * from '#components/stock_transfers/StockTransfer'
export * from '#components/stock_transfers/StockTransferField'
export * from '#typings/errors'
