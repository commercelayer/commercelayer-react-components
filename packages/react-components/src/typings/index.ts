import { type Dispatch, type ForwardedRef, type Ref } from 'react'
import PropTypes, { type InferProps } from 'prop-types'
import { type BaseError } from './errors'

export const BC = {
  id: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object,
  name: PropTypes.string
}

export const PTLoader = PropTypes.oneOfType([
  PropTypes.element,
  PropTypes.string
])

export const BaseSelectComponentPropTypes = {
  children: PropTypes.func,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
        .isRequired,
      selected: PropTypes.bool
    }).isRequired
  ).isRequired,
  placeholder: PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
  }),
  value: PropTypes.string,
  name: PropTypes.string.isRequired
}

export type SelectPlaceholder = Option

type BaseSelectChildrenComponentProps = Omit<
  BaseSelectComponentProps,
  'children'
>

interface Option {
  label: string
  value: string | number
  disabled?: boolean
}

export interface BaseSelectComponentProps {
  children?: (props: BaseSelectChildrenComponentProps) => JSX.Element
  options: Option[]
  placeholder?: Option
  value?: string
  className?: string
  required?: boolean
  name: string
  onChange?: (event: React.ChangeEvent<HTMLSelectElement>) => void
}

type BaseInputChildrenComponentProps = Omit<
  BaseInputComponentProps,
  'children'
> & {
  handleChange: (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => void
  parentRef: ForwardedRef<any>
}

export interface BaseInputComponentProps {
  ref?: Ref<any>
  children?: (props: BaseInputChildrenComponentProps) => JSX.Element
  name: string
  onChange?: (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => void
  placeholder?: string
}

export type GiftCardInputName =
  | 'balanceCents'
  | 'balanceMaxCents'
  | 'singleUse'
  | 'rechargeable'
  | 'imageUrl'
  | 'expiresAt'
  | 'referenceOrigin'
  | 'email'
  | 'firstName'
  | 'lastName'
  | 'reference'

export type AddressInputName =
  | 'billing_address_city'
  | 'billing_address_company'
  | 'billing_address_first_name'
  | 'billing_address_email'
  | 'billing_address_last_name'
  | 'billing_address_line_1'
  | 'billing_address_line_2'
  | 'billing_address_phone'
  | 'billing_address_state_code'
  | 'billing_address_zip_code'
  | 'billing_address_billing_info'
  | 'billing_address_save_to_customer_book'
  | `billing_address_${`metadata_${string}`}`
  | 'shipping_address_city'
  | 'shipping_address_company'
  | 'shipping_address_email'
  | 'shipping_address_first_name'
  | 'shipping_address_last_name'
  | 'shipping_address_line_1'
  | 'shipping_address_line_2'
  | 'shipping_address_phone'
  | 'shipping_address_state_code'
  | 'shipping_address_zip_code'
  | 'shipping_address_save_to_customer_book'

export type AddressCountrySelectName =
  | 'billing_address_country_code'
  | 'shipping_address_country_code'

export type AddressStateSelectName =
  | 'billing_address_state_code'
  | 'shipping_address_state_code'

export type LoaderType = string | JSX.Element

export const BMObject = PropTypes.objectOf(PropTypes.string)
export type BaseMetadataObject = Record<string, string | undefined | null>

export type TimeFormat = 'days' | 'hours'

export type BaseComponent = InferProps<typeof BC>

export interface BaseAction<A = string, P = Record<string, any>> {
  type: A
  payload: P
}

export interface BaseState {
  [key: string]: any
  errors?: BaseError[]
}

export type BaseActionType<T = string> = T[]

export type BFSetStateContainer<T> = <P extends T>(param: P) => void

export type BaseReducer = <
  S extends BaseState,
  A extends BaseAction,
  T extends BaseActionType
>(
  state: S,
  action: A,
  type: T
) => S

export type BaseUnsetState<A> = (dispatch: Dispatch<A>) => void

export type BaseMetadata = Record<string, string | undefined | null>

export type BaseFormatPrice = 'formatted' | 'cents' | 'float'

export const baseOrderPricePropTypes = {
  base: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  children: PropTypes.func,
  format: PropTypes.oneOf<BaseFormatPrice>(['formatted', 'cents', 'float']),
  ...BC
}

export const baseOrderComponentPricePropTypes = {
  children: baseOrderPricePropTypes.children,
  format: baseOrderPricePropTypes.format,
  ...BC
}

export type BasePriceType = 'total' | 'option' | 'unit'
export type BaseSelectorType = 'select' | 'radio'

export type BaseAmountComponentChildren = Omit<BaseAmountComponent, 'children'>

export interface BaseAmountComponent
  extends Omit<JSX.IntrinsicElements['span'], 'children'> {
  children?: ChildrenFunction<BaseAmountComponentChildren>
  format?: BaseFormatPrice
  price?: string
  scheme?: string
  priceCents?: number
  labelFree?: string
}

export type ChildrenFunction<P = Record<string, any>> = (
  props: P
) => JSX.Element | null

export type ExcludeTag<T extends keyof JSX.IntrinsicElements> = Exclude<
  keyof JSX.IntrinsicElements,
  T
>

export type ExtractTag<T extends keyof JSX.IntrinsicElements> = Extract<
  keyof JSX.IntrinsicElements,
  T
>

export type ConditionalElement<E> =
  | ({
      attribute: Extract<keyof E, 'image_url'>
      tagElement?: ExtractTag<'img'>
    } & JSX.IntrinsicElements['img'])
  | ({
      attribute: Exclude<keyof E, 'image_url'>
      tagElement?: ExcludeTag<'img'>
    } & Omit<JSX.IntrinsicElements[ExcludeTag<'img'>], 'children'>)
