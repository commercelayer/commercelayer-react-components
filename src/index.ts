export { default as CommerceLayer } from './components/CommerceLayer'
export { default as OrderContainer } from './components/OrderContainer'
export { default as PricesContainer } from './components/PricesContainer'
export { default as Price } from './components/Price'
export { default as VariantsContainer } from './components/VariantsContainer'
export { default as VariantSelector } from './components/VariantSelector'
export { default as QuantitySelector } from './components/QuantitySelector'
export { default as AddToCartButton } from './components/AddToCartButton'
export { default as AvailabilityContainer } from './components/AvailabilityContainer'
export { default as AvailabilityTemplate } from './components/AvailabilityTemplate'
export { default as GiftCardContainer } from './components/GiftCardContainer'
export { default as GiftCard } from './components/GiftCard'
export { default as GiftCardCurrencySelector } from './components/GiftCardCurrencySelector'
export { default as GiftCardInput } from './components/GiftCardInput'
export { default as MetadataInput } from './components/MetadataInput'
export { default as SubmitButton } from './components/SubmitButton'
export { default as ItemContainer } from './components/ItemContainer'
export { default as LineItemsCount } from './components/LineItemsCount'
export { default as LineItemsContainer } from './components/LineItemsContainer'
export { default as LineItem } from './components/LineItem'
export { default as LineItemImage } from './components/LineItemImage'
export { default as LineItemName } from './components/LineItemName'
export { default as LineItemAmount } from './components/LineItemAmount'
export { default as LineItemQuantity } from './components/LineItemQuantity'
export { default as LineItemRemoveLink } from './components/LineItemRemoveLink'
export { default as SubTotalAmount } from './components/SubTotalAmount'
export { default as DiscountAmount } from './components/DiscountAmount'
export { default as ShippingAmount } from './components/ShippingAmount'
export { default as TaxesAmount } from './components/TaxesAmount'
export { default as GiftCardAmount } from './components/GiftCardAmount'
export { default as TotalAmount } from './components/TotalAmount'
export { default as CheckoutLink } from './components/CheckoutLink'
export { default as Errors } from './components/Errors'
export { default as SkuOptionsContainer } from './components/SkuOptionsContainer'
export { default as SkuOption } from './components/SkuOption'
export { default as SkuOptionInput } from './components/SkuOptionInput'
export { default as LineItemOptions } from './components/LineItemOptions'
export { default as LineItemOption } from './components/LineItemOption'
export { default as SkuListsContainer } from './components/SkuListsContainer'
export { default as SkuList } from './components/SkuList'
