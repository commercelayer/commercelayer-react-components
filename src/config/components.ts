import PropTypes from 'prop-types'
import childrenTypes from '../utils/childrenTypes'
import {
  TimeFormat,
  BaseInputType,
  GiftCardInputName,
  baseOrderComponentPricePropTypes,
  LineItemType,
  BasePriceType,
  BaseFormatPrice,
  PTLoader,
  BMObject,
  BaseSelectorType,
} from '../typings'
import { ErrorPropTypes } from '../typings/errors'
import { BaseInputComponentPropTypes } from '../typings/index'
import SkuList from '../components/SkuList'

const components = {
  AddToCartButton: {
    displayName: 'AddToCartButton',
    propTypes: {
      children: PropTypes.func,
      label: PropTypes.string,
      skuCode: PropTypes.string,
      disabled: PropTypes.bool,
    },
    defaultProps: {
      label: 'Add to cart',
    },
  },
  AvailabilityContainer: {
    displayName: 'AvailabilityContainer',
    permittedChildren: ['AvailabilityTemplate', 'ReactNode'],
    propTypes: {
      children: childrenTypes.isRequired,
      skuCode: PropTypes.string,
    },
  },
  AvailabilityTemplate: {
    displayName: 'AvailabilityTemplate',
    propTypes: {
      timeFormat: PropTypes.oneOf<TimeFormat>(['days', 'hours']),
      showShippingMethodName: PropTypes.bool,
      children: PropTypes.func,
    },
    defaultProps: {
      timeFormat: 'days',
      showShippingMethodName: false,
    },
  },
  CheckoutLink: {
    displayName: 'CheckoutLink',
    propTypes: {
      children: PropTypes.func,
      label: PropTypes.string,
    },
    defaultProps: {
      label: 'Checkout',
    },
  },
  CommerceLayer: {
    permittedChildren: [
      'OrderContainer',
      'PricesContainer',
      'GiftCardContainer',
      // 'AvailabilityContainer',
      'ReactNode',
    ],
    propTypes: {
      children: childrenTypes.isRequired,
      accessToken: PropTypes.string.isRequired,
      endpoint: PropTypes.string.isRequired,
    },
  },
  DiscountAmount: {
    displayName: 'DiscountAmount',
    propTypes: baseOrderComponentPricePropTypes,
    defaultProps: {
      format: 'formatted' as BaseFormatPrice,
    },
  },
  Errors: {
    displayName: 'Errors',
    propTypes: ErrorPropTypes,
    defaultProps: {
      messages: [],
      field: 'base',
    },
  },
  GiftCard: {
    permittedChildren: [
      'GiftCardCurrencySelector',
      'GiftCardInput',
      'Errors',
      'MetadataInput',
      'SubmitButton',
      'ReactNode',
    ],
    displayName: 'GiftCard',
    propTypes: {
      children: childrenTypes.isRequired,
      onSubmit: PropTypes.func,
    },
    defaultProps: {
      onSubmit: undefined,
    },
  },
  GiftCardAmount: {
    displayName: 'GiftCardAmount',
    propTypes: baseOrderComponentPricePropTypes,
  },
  GiftCardContainer: {
    permittedChildren: ['GiftCard', 'Errors', 'ReactNode'],
    displayName: 'GiftCardContainer',
    propTypes: {
      children: childrenTypes.isRequired,
    },
  },
  GiftCardCurrencySelector: {
    displayName: 'GiftCardCurrencySelector',
    propTypes: {
      children: PropTypes.func,
      placeholder: PropTypes.shape({
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
          .isRequired,
        label: PropTypes.string.isRequired,
      }),
      value: PropTypes.string,
      required: PropTypes.bool,
    },
    defaultProps: {
      required: true,
    },
  },
  GiftCardInput: {
    displayName: 'GiftCardInput',
    propTypes: {
      type: PropTypes.oneOf<BaseInputType>([
        'text',
        'email',
        'number',
        'date',
        'checkbox',
      ]).isRequired,
      name: PropTypes.oneOf<GiftCardInputName>([
        'balanceCents',
        'balanceMaxCents',
        'singleUse',
        'rechargeable',
        'imageUrl',
        'expiresAt',
        'referenceOrigin',
        'email',
        'firstName',
        'lastName',
        'reference',
      ]).isRequired,
      children: PropTypes.func,
      placeholder: PropTypes.string,
    },
  },
  ItemContainer: {
    permittedChildren: [
      'PricesContainer',
      'VariantsContainer',
      'SkuOptionContainer',
      'QuantitySelector',
      'AddToCartButton',
      'AvailabilityContainer',
      'SkuListsContainer',
      'ReactNode',
    ],
    displayName: 'ItemContainer',
    propTypes: {
      children: childrenTypes.isRequired,
      skuCode: PropTypes.string,
      lineItem: PropTypes.shape({
        name: PropTypes.string.isRequired,
        imageUrl: PropTypes.string,
      }),
    },
  },
  LineItem: {
    permittedChildren: [
      'LineItemImage',
      'LineItemName',
      'LineItemOptions',
      'LineItemQuantity',
      'LineItemAmount',
      'LineItemRemoveLink',
      'Errors',
      'ReactNode',
    ],
    displayName: 'LineItem',
    propTypes: {
      children: childrenTypes.isRequired,
      type: PropTypes.oneOf<LineItemType>([
        'skus',
        'gift_cards',
        'shipments',
        'paymentMethods',
        'promotions',
      ]),
    },
    defaultProps: {
      type: 'skus',
    },
  },
  LineItemAmount: {
    displayName: 'LineItemAmount',
    propTypes: {
      ...baseOrderComponentPricePropTypes,
      type: PropTypes.oneOf<BasePriceType>(['total', 'unit', 'option']),
    },
    defaultProps: {
      format: 'formatted',
      type: 'total',
    },
  },
  LineItemImage: {
    displayName: 'LineItemImage',
    propTypes: {
      width: PropTypes.number,
      children: PropTypes.func,
    },
  },
  LineItemName: {
    displayName: 'LineItemName',
    propTypes: {
      children: PropTypes.func,
    },
  },
  LineItemOption: {
    displayName: 'LineItemOption',
    propTypes: {
      name: PropTypes.string.isRequired,
      children: PropTypes.func,
      valueClassName: PropTypes.string,
      keyClassName: PropTypes.string,
      keyId: PropTypes.string,
      keyStyle: PropTypes.object,
    },
  },
  LineItemOptions: {
    permittedChildren: ['LineItemOption', 'ReactNode'],
    displayName: 'LineItemOptions',
    propTypes: {
      children: childrenTypes.isRequired,
      title: PropTypes.string,
      showName: PropTypes.bool,
      skuOptionId: PropTypes.string.isRequired,
    },
    defaultProps: {
      showName: true,
    },
  },
  LineItemQuantity: {
    displayName: 'LineItemQuantity',
    propTypes: {
      children: PropTypes.func,
      max: PropTypes.number,
      disabled: PropTypes.bool,
    },
    defaultProps: {
      max: 50,
    },
  },
  LineItemRemoveLink: {
    displayName: 'LineItemRemoveLink',
    propTypes: {
      children: PropTypes.func,
      label: PropTypes.string,
    },
    defaultProps: {
      label: 'Remove',
    },
  },
  LineItemsContainer: {
    permittedChildren: ['LineItemsCount', 'LineItem', 'ReactNode'],
    displayName: 'LineItemsContainer',
    propTypes: {
      children: childrenTypes.isRequired,
      filters: PropTypes.object,
      loader: PTLoader,
    },
    defaultProps: {
      filters: {},
      loader: 'Loading...',
    },
  },
  LineItemsCount: {
    displayName: 'LineItemsCount',
    propTypes: {
      children: PropTypes.func,
      id: PropTypes.string,
      className: PropTypes.string,
      name: PropTypes.string,
      style: PropTypes.object,
    },
  },
  MetadataInput: {
    displayName: 'MetadataInput',
    propTypes: BaseInputComponentPropTypes,
  },
  OrderContainer: {
    permittedChildren: [
      'ItemContainer',
      'LineItemsContainer',
      'SubTotalAmount',
      'DiscountAmount',
      'ShippingAmount',
      'TaxesAmount',
      'GiftCardAmount',
      'TotalAmount',
      'CheckoutLink',
      'GiftCardContainer',
      'ReactNode',
    ],
    displayName: 'OrderContainer',
    propTypes: {
      children: childrenTypes.isRequired,
      persistKey: PropTypes.string.isRequired,
      metadata: BMObject,
    },
    defaultProps: {
      metadata: {},
    },
  },
  Price: {
    displayName: 'Price',
    propTypes: {
      children: PropTypes.func,
      compareClassName: PropTypes.string,
      skuCode: PropTypes.string,
      showCompare: PropTypes.bool,
    },
    defaultProps: {
      skuCode: '',
    },
  },
  PricesContainer: {
    permittedChildren: ['Price', 'ReactNode'],
    displayName: 'PricesContainer',
    propTypes: {
      children: childrenTypes.isRequired,
      skuCode: PropTypes.string,
      loader: PTLoader,
      perPage: PropTypes.number,
      filters: PropTypes.object,
    },
    defaultProps: {
      perPage: 10,
      filters: {},
      loader: 'Loading...',
      skuCode: '',
    },
  },
  QuantitySelector: {
    displayName: 'QuantitySelector',
    propTypes: {
      children: PropTypes.func,
      min: PropTypes.number,
      max: PropTypes.number,
      value: PropTypes.string,
      skuCode: PropTypes.string,
      disabled: PropTypes.bool,
    },
    defaultProps: {
      min: 1,
    },
  },
  ShippingAmount: {
    displayName: 'ShippingAmount',
    propTypes: baseOrderComponentPricePropTypes,
  },
  SkuList: {
    permittedChildren: ['AddToCartButton', 'QuantitySelector', 'ReactNode'],
    displayName: 'SkuList',
    propTypes: {
      children: childrenTypes.isRequired,
      id: PropTypes.string.isRequired,
    },
  },
  SkuListsContainer: {
    permittedChildren: ['SkuList', 'ReactNode'],
    displayName: 'SkuListsContainer',
    propTypes: {
      children: childrenTypes.isRequired,
    },
  },
  SkuOption: {
    permittedChildren: ['SkuOptionInput', 'ReactNode'],
    displayName: 'SkuOption',
    propTypes: {
      children: childrenTypes.isRequired,
      id: PropTypes.string.isRequired,
    },
  },
  SkuOptionInput: {
    displayName: 'SkuOptionInput',
    propTypes: BaseInputComponentPropTypes,
  },
  SkuOptionsContainer: {
    permittedChildren: ['SkuOption', 'ReactNode'],
    displayName: 'SkuOptionsContainer',
    propTypes: {
      children: childrenTypes.isRequired,
      skuCode: PropTypes.string,
    },
  },
  SubmitButton: {
    displayName: 'SubmitButton',
    propTypes: {
      children: PropTypes.func,
      label: PropTypes.string,
    },
    defaultProps: {
      label: 'Submit',
    },
  },
  SubTotalAmount: {
    displayName: 'SubTotalAmount',
    propTypes: baseOrderComponentPricePropTypes,
    defaultProps: {
      format: 'formatted' as BaseFormatPrice,
    },
  },
  TaxesAmount: {
    displayName: 'TaxesAmount',
    propTypes: baseOrderComponentPricePropTypes,
    defaultProps: {
      format: 'formatted' as BaseFormatPrice,
    },
  },
  TotalAmount: {
    displayName: 'TotalAmount',
    propTypes: baseOrderComponentPricePropTypes,
    defaultProps: {
      format: 'formatted' as BaseFormatPrice,
    },
  },
  VariantsContainer: {
    permittedChildren: ['VariantSelector', 'ReactNode'],
    displayName: 'VariantsContainer',
    propTypes: {
      children: childrenTypes.isRequired,
      skuCode: PropTypes.string,
      filters: PropTypes.object,
    },
    defaultProps: {
      skuCode: '',
      filters: {},
    },
  },
  VariantSelector: {
    displayName: 'VariantSelector',
    propTypes: {
      options: PropTypes.arrayOf(
        PropTypes.shape({
          label: PropTypes.string.isRequired,
          code: PropTypes.string.isRequired,
          lineItem: PropTypes.shape({
            name: PropTypes.string.isRequired,
            imageUrl: PropTypes.string,
          }),
        }).isRequired
      ).isRequired,
      name: PropTypes.string,
      children: PropTypes.func,
      type: PropTypes.oneOf<BaseSelectorType>(['select', 'radio']),
      loader: PropTypes.element,
      placeholder: PropTypes.string,
      skuCode: PropTypes.string,
    },
    defaultProps: {
      placeholder: 'Select a variant',
      type: 'select' as BaseSelectorType,
    },
  },
}

export default components
