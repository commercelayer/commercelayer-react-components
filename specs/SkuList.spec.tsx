import React from 'react'
import { SkuList, Price } from '../src'
import renderer from 'react-test-renderer'
import components from '../src/config/components'

const propTypes = components.SkuList.propTypes

test('<SkuList/>', () => {
  expect.assertions(3)
  const component = renderer.create(
    <SkuList id="unit-test">
      <div>test</div>
    </SkuList>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.id).toBe(propTypes.id)
})

test('<SkuList proptypes required />', () => {
  expect.assertions(3)
  console.error = jest.fn()
  const component = renderer.create(<SkuList />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  expect(console.error.mock.calls[0][0]).toEqual(
    expect.stringContaining(
      `Warning: Failed prop type: The prop 'children' is marked as required in 'SkuList', but its value is 'undefined'.`
    )
  )
  expect(console.error.mock.calls[1][0]).toEqual(
    expect.stringContaining(
      'Warning: Failed prop type: The prop `id` is marked as required in `SkuList`, but its value is `undefined`.'
    )
  )
})

test('<SkuList check children />', () => {
  expect.assertions(2)
  console.error = jest.fn()
  const component = renderer.create(
    <SkuList id="unit-id">
      <Price />
    </SkuList>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  expect(console.error.mock.calls[0][0]).toEqual(
    expect.stringContaining(
      `Warning: Failed prop type: Invalid prop 'children' supplied to SkuList. Only components AddToCartButton, QuantitySelector, ReactNode are allowed.`
    )
  )
})
