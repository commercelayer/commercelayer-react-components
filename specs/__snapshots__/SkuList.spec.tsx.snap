// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkuList check children /> 1`] = `null`;

exports[`<SkuList proptypes required /> 1`] = `null`;

exports[`<SkuList/> 1`] = `
<div>
  test
</div>
`;
